src/libraries/System.Console/tests/ManualTests/ManualTests.cs

```diff
@@ -321,6 +323,44 @@ public static void EncodingTest()
             AssertUserExpectedResults("Pi and Sigma or question marks");
         }
 
+        [ConditionalFact(nameof(ManualTestsEnabled))]
+        [SkipOnPlatform(TestPlatforms.Browser | TestPlatforms.iOS | TestPlatforms.MacCatalyst | TestPlatforms.tvOS, "Not supported on Browser, iOS, MacCatalyst, or tvOS.")]
+        public static void ResizeTest()
+        {
+            bool wasResized = false;
+
+            using (ManualResetEvent manualResetEvent = new(false))
+            using (PosixSignalRegistration.Create(PosixSignal.SIGWINCH,
+                       ctx =>
+                       {
+                           wasResized = true;
+                           Assert.Equal(PosixSignal.SIGWINCH, ctx.Signal);
+                           manualResetEvent.Set();
+                       }))
+            {
+                int widthBefore = Console.WindowWidth;
+                int heightBefore = Console.WindowHeight;
+
+                Assert.False(wasResized);
+
+                Console.SetWindowSize(widthBefore / 2, heightBefore / 2);
+
+                Assert.True(manualResetEvent.WaitOne(TimeSpan.FromMilliseconds(50)));
+                Assert.True(wasResized);
+                Assert.Equal(widthBefore / 2, Console.WindowWidth );
+                Assert.Equal(heightBefore / 2, Console.WindowHeight );
+
+                try
+                {
+                    AssertUserExpectedResults("terminal window getting two times smaller?");
+                }
+                finally
+                {
+                    Console.SetWindowSize(widthBefore, heightBefore);
+                }
```

Comment:
Please excuse me, my assumptions about the visible effects of this action were wrong.
```suggestion
                Console.SetWindowSize(widthBefore, heightBefore);
```